### PR TITLE
Docs: DO-2133 update docs for new action api

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

With the new API for actions out a lot of examples in our docs were using the deprecated version

## Implementation Description

Updated examples throughout the docs to use the new action API

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [ ] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

**Review by tamara-causalens — APPROVED** [review]

File: docs/best-practices/app-structure-breakdown.md [hunk 1]
```diff
@@ -170,7 +170,7 @@ Your app will ultimately perform some logic based on how the user interacts with
 
 ### Resolver Functions
 
-When the user interacts with the app and updates the values of the `Variable`s in your app, this will update `DerivedVariable`s and `DataDerivedVariable`s or trigger actions like `dara.core.interactivity.actions.UpdateVariable` and `dara.core.interactivity.actions.SideEffect`. What these have in common is they take a resolver function that defines how to update based on the new values of the `Variable`s.
+When the user interacts with the app and updates the values of the `Variable`s in your app, this will update `DerivedVariable`s and `DataDerivedVariable`s or trigger actions like `dara.core.interactivity.actions.
```
Comment (krzysztof-causalens) [inline docs/best-practices/app-structure-breakdown.md:173]: I don't think we need the `dara.core.interactivity.actions` link anymore, probably just link to the actions docs page instead?

File: docs/getting-started/handling-data.mdx [hunk 2]
```diff
@@ -65,39 +65,41 @@ my_first_data_variable = DataVariable(data, cache=CacheType.GLOBAL)
...
 
 target = Variable()
 
-UpdateVariable(
-    my_resolver_function,
-    variable=target,
-    extras=[my_first_data_variable]
-)
+@action
+async def my_resolver_function(ctx: action.Ctx, my_data_var):
```
Comment (krzysztof-causalens) [inline docs/getting-started/handling-data.mdx:80]: I think `my_data_var` name here can be confusing, can we rename (call it `...value` or something?) and give it a type annotation of `pandas.DataFrame` to be explicit it's the value?

File: docs/getting-started/handling-data.mdx [hunk 3]
```diff
@@ -65,39 +65,41 @@ my_first_data_variable = DataVariable(data, cache=CacheType.GLOBAL)
...
+    # my_data_var here takes the value of my_first_data_variable
+    ...
+    ctx.update(variable=target, value=my_data_var)
```
Comment (krzysztof-causalens) [inline docs/getting-started/handling-data.mdx:83]: missing `await`

File: docs/packages/dara-components/graphs/introduction.mdx [hunk 4]
```diff
@@ -284,13 +283,16 @@ causal_graph.add_edge('A', 'B')
...
-def resolver_on_click_node(ctx: UpdateVariable.Ctx):
-    if isinstance(ctx.inputs.new, dict):
-        return ctx.inputs.new.get('identifier')
-    return None
+@action
+async def resolver_on_click_node(ctx: action.Ctx):
+    if isinstance(ctx.input, dict):
+        await ctx.update(variable=selected_node, value=ctx.input.get('identifier'))
+    else:
+        await ctx.update(variable=selected_node, value=None)
```
Comment (krzysztof-causalens) [inline docs/packages/dara-components/graphs/introduction.mdx:291]: This can be simplified to e.g.
```python
value = ctx.input.get('identifier') if isinstance(ctx.input, dict) else None
await ctx update(...)
```

File: docs/packages/dara-components/plotting/bokeh_events.md [hunk 5]
```diff
@@ -114,46 +125,62 @@ def interactive_wheat_scatter_plot():
...
         )
     else:
         return Stack(
-            Text(f'The selected point is of the {selected_variety} wheat variety.'),
+            Text(f"The selected point is of the {selected_variety} wheat variety."),
             Button(
-                f'Read More on Triticum {selected_variety}',
-                onclick=NavigateTo(url=f'https://en.wikipedia.org/wiki/Triticum_{selected_variety}', new_tab=True),
+                f"Read More on Triticum {selected_variety}",
+                onclick=navigate_to(selected_variety),
```
Comment (krzysztof-causalens) [inline docs/packages/dara-components/plotting/bokeh_events.md:166]: This example can use a static url, e.g. `NavigateTo(f'https://(...){selected_variety}')`, no need to make a function

File: packages/dara-components/dara/components/common/button.py [hunk 6]
```diff
@@ -49,14 +49,16 @@ class Button(LayoutComponent):
...
 
-    action = NavigateTo(url='/test')
+    @action
+    async def navigate_to(ctx: action.Ctx):
+        await ctx.navigate(url='/test')
 
     Button(
         'Click',
-        onclick=action,
+        onclick=navigate_to(),
```
Comment (krzysztof-causalens) [inline packages/dara-components/dara/components/common/button.py:61]: Same here and below, with a static url just construct `NavigateTo` object directly with it e.g. `onclick=NavigateTo(url)`. Otherwise we need to call to the backend to get the url

File: packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py [hunk 7]
```diff
@@ -106,13 +105,16 @@ class CausalGraphViewer(BaseGraphComponent):
...
-    def resolver_on_click_node(ctx: UpdateVariable.Ctx):
-        if isinstance(ctx.inputs.new, dict):
-            return ctx.inputs.new.get('identifier')
-        return None
+    @action
+    async def resolver_on_click_node(ctx: action.Ctx):
+        if isinstance(ctx.input, dict):
+            await ctx.update(variable=selected_node, value=ctx.input.get('identifier'))
+        else:
+            await ctx.update(variable=selected_node, value=None)
```
Comment (krzysztof-causalens) [inline packages/dara-components/dara/components/graphs/components/causal_graph_viewer.py:113]: Can simplify by computing value first, just like in the docs

**Review by krzysztof-causalens — COMMENTED** [review]
Nice job! A few nitpicks in places we could improve but looks good overall

File: docs/getting-started/handling-data.mdx [hunk 8]
```diff
@@ -65,39 +65,41 @@ my_first_data_variable = DataVariable(data, cache=CacheType.GLOBAL)
...
 ### Updating
 
 You can update the data stored in a `DataVariable` by:
 
-- using the `dara.core.interactivity.actions.UpdateVariable` action
+- using a `dara.core.interactivity.actions` action
 
 ```python
-from dara.core import DataVariable, CacheType, UpdateVariable
+from dara.core import DataVariable, CacheType
```
Comment (claudiawestby) [inline docs/getting-started/handling-data.mdx:95]: minor but missing pandas import i think

File: docs/packages/dara-components/plotting/bokeh_events.md [hunk 9]
```diff
@@ -68,10 +67,10 @@ Now to use this event with a `Bokeh` component, the component has to register ac
 
 ```python
     ...
-    # the `new` parameter of the UpdateVariable resolver will hold the value returned by the callback
+    # we sync the value of the selected_variety_var to the value returned by the callback
```
Comment (claudiawestby) [inline docs/packages/dara-components/plotting/bokeh_events.md:70]: use of first person "we"

**Review by claudiawestby — APPROVED** [review]
really nice job!

File: docs/getting-started/handling-data.mdx (hunk 8 above)
Comment (patricia-causalens) [reply docs/getting-started/handling-data.mdx:96]: good catch, fixed

File: packages/dara-components/dara/components/common/button.py (hunk 6 above)
Comment (patricia-causalens) [reply packages/dara-components/dara/components/common/button.py]: fixed

File: docs/packages/dara-components/graphs/introduction.mdx (hunk 4 above)
Comment (patricia-causalens) [reply docs/packages/dara-components/graphs/introduction.mdx]: good point, updated

File: docs/getting-started/handling-data.mdx (hunk 3 above)
Comment (patricia-causalens) [reply docs/getting-started/handling-data.mdx]: I knew there had to be one missing somewhere, fixed!

File: docs/best-practices/app-structure-breakdown.md (hunk 1 above)
Comment (patricia-causalens) [reply docs/best-practices/app-structure-breakdown.md]: good point

**Review by krzysztof-causalens — APPROVED** [review]
